src/Ocelot/Configuration/Creator/RouteOptionsCreator.cs

```diff
@@ -27,7 +27,12 @@ public RouteOptions Create(FileRoute fileRoute)
 
         private static bool IsEnableRateLimiting(FileRoute fileRoute) => fileRoute.RateLimitOptions?.EnableRateLimiting == true;
 
-        private static bool IsAuthenticated(FileRoute fileRoute) => !string.IsNullOrEmpty(fileRoute.AuthenticationOptions?.AuthenticationProviderKey);
+        private static bool IsAuthenticated(FileRoute fileRoute, FileGlobalConfiguration globalConfiguration)
+        {
+            return (!string.IsNullOrEmpty(globalConfiguration?.AuthenticationOptions?.AuthenticationProviderKey) &&
+                                                              !fileRoute.AuthenticationOptions.AllowAnonymousForGlobalAuthenticationOptions) ||
+                !string.IsNullOrEmpty(fileRoute.AuthenticationOptions?.AuthenticationProviderKey);
+        }

```

Comment:
Because the body consists of one operator only, why not to define via expression body 👇 
```suggestion
        private static bool IsAuthenticated(FileRoute fileRoute, FileGlobalConfiguration globalConfiguration)
            => (!string.IsNullOrEmpty(globalConfiguration?.AuthenticationOptions?.AuthenticationProviderKey)
                && !fileRoute.AuthenticationOptions.AllowAnonymousForGlobalAuthenticationOptions)
                || !string.IsNullOrEmpty(fileRoute.AuthenticationOptions?.AuthenticationProviderKey);
```